condor_history_to_prometheus.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+import os, sys
+import glob
+from optparse import OptionParser
+import logging
+import htcondor, classad
+from condor_utils import *
+from condor_job_metrics import JobMetrics
+import datetime
+import time
+import prometheus_client
+from datetime import datetime
+import re
+
+utc_format = '%Y-%m-%dT%H:%M:%S'
+
+def generate_ads(entries):
+    for data in entries:
+        add_classads(data)
+        yield data
+
+def compose_ad_metrics(ad, metrics):
+    ''' Parse condor job classad and update metrics
+
+        Args:
+            ad (classad): an HTCondor job classad
+            metrics (JobMetrics): JobMetrics object 
+    '''
+    labels = {'owner': None,
+              'site': None,
+              'schedd': None,
+              'GPUDeviceName': None,
+              'usage': None,
+              'kind': None}
+
+    labels['owner'] = ad['Owner']
+    labels['site'] = ad['MATCH_EXP_JOBGLIDEIN_ResourceName']
+    labels['schedd'] = ad['GlobalJobId'][0:ad['GlobalJobId'].find('#')]
+    labels['GPUDeviceName'] = None
+
+    # ignore this ad if walltimehrs is negative
+    if ad['walltimehrs'] < 0:
+        return
+    
+    metrics.condor_job_walltime_hours.labels(**labels).inc(ad['walltimehrs'])
```

Comment:
Shouldn't this come after the rest of the labels? It might be useful to be able to separate out walltimes for CPU vs GPU jobs.

Reply:
I had put it before setting other labels to ignore the type, but I suppose it can still just be summed to get that same value

Reply:
Yeah, summing was what I was thinking.  This means the monitoring data is "richer" in case we want to do things later with it.  And there isn't this weird edge case in one of the metric series.